cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/batch/MapReduceLifecycleContext.java

```diff
@@ -0,0 +1,210 @@
+    this.delegate = delegate;
+  }
+
+  @Override
+  public void close() throws IOException {
+    delegate.close();
+  }
+
+  @Override
+  public <K, V> void write(String namedOutput, K key, V value) throws IOException, InterruptedException {
+    delegate.write(namedOutput, key, value);
+  }
+
+  @Override
+  public void write(KEYOUT key, VALUEOUT value) throws IOException, InterruptedException {
+    delegate.write(key, value);
+  }
+
+  @Override
+  public MapReduceSpecification getSpecification() {
+    return delegate.getSpecification();
+  }
+
+  @Override
+  public long getLogicalStartTime() {
+    return delegate.getLogicalStartTime();
+  }
+
+  @Override
+  public <T> T getHadoopContext() {
+    return delegate.getHadoopContext();
+  }
+
+  @Nullable
+  @Override
+  public WorkflowToken getWorkflowToken() {
+    return delegate.getWorkflowToken();
+  }
+
+  @Nullable
+  @Override
+  public AdapterSpecification getAdapterSpecification() {
+    return delegate.getAdapterSpecification();
+  }
+
+  @Override
+  public PluginProperties getPluginProperties(String pluginId) {
+    return delegate.getPluginProperties(pluginId);
+  }
+
+  @Override
+  public <T> Class<T> loadPluginClass(String pluginId) {
+    return delegate.loadPluginClass(pluginId);
+  }
+
+  @Override
+  public <T> T newPluginInstance(String pluginId) throws InstantiationException {
+    return null;
+  }
+
+  @Override
+  public <T extends Dataset> T getDataset(String name) throws DatasetInstantiationException {
+    return delegate.getDataset(name);
+  }
+
+  @Override
+  public <T extends Dataset> T getDataset(String name,
+                                          Map<String, String> arguments) throws DatasetInstantiationException {
+    return delegate.getDataset(name, arguments);
+  }
+
+  @Override
+  public PluginProperties getPluginProps(String pluginId) {
+    return delegate.getPluginProps(pluginId);
+  }
+
+  @Override
+  public <T> Class<T> loadClass(String pluginId) {
+    return delegate.loadClass(pluginId);
+  }
+
+  @Override
+  public <T> T newInstance(String pluginId) throws InstantiationException {
+    return delegate.newInstance(pluginId);
+  }
+
+  @Override
+  public Map<String, String> getRuntimeArguments() {
+    return delegate.getRuntimeArguments();
+  }
+
+  @Override
+  public URL getServiceURL(String applicationId, String serviceId) {
+    return delegate.getServiceURL(applicationId, serviceId);
+  }
+
+  @Override
+  public URL getServiceURL(String serviceId) {
+    return delegate.getServiceURL(serviceId);
+  }
+
+  // TODO: Document usage of ProgramLifeCycle<MapReduceTaskContext> instead of ProgramLifeCycle<MapReduceContext>
+  // Until then, we can not remove the following methods.
+  private static final String UNSUPPORTED_ERROR_MESSAGE = "Operation not supported from MapReduce task-level context.";
+
+  @Override
+  public <T> T getHadoopJob() {
+    throw new UnsupportedOperationException(UNSUPPORTED_ERROR_MESSAGE);
```

Comment:
Instead of throwing exception, which can bring existing program, better log a warn and do no-op.